### ISSUE TITLE
Broken routing on some networks

### ISSUE THREAD

**Issue by janjaromirhorak** [janjaromirhorak/hradla#16]
Routing does not work for this nework:

```JSON
{
  "boxes": [
    {
      "name": "input",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [36, 87]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 0},
        {"index": 0, "wireId": 1}
      ],
      "isOn": false
    },
    {
      "name": "input",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [36, 98]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 2},
        {"index": 0, "wireId": 3}
      ],
      "isOn": false
    },
    {
      "name": "input",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [36, 109]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 4},
        {"index": 0, "wireId": 5}
      ],
      "isOn": false
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {
            "name": "translate",
            "args": [77, 108]
          },
          {"name": "rotate", "args": ["0", 1, 4]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 6},
        {"index": 1, "wireId": 7},
        {"index": 2, "wireId": 8}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 107]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 9},
        {"index": 1, "wireId": 10},
        {"index": 2, "wireId": 6}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 96]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 11},
        {"index": 1, "wireId": 12},
        {"index": 2, "wireId": 13}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 97]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 13},
        {"index": 1, "wireId": 14},
        {"index": 2, "wireId": 15}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 85]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 16},
        {"index": 1, "wireId": 17},
        {"index": 2, "wireId": 18}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 86]},
          {"name": "rotate", "args": ["0", 1, 2]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 18},
        {"index": 1, "wireId": 19},
        {"index": 2, "wireId": 20}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 74]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 21},
        {"index": 1, "wireId": 22},
        {"index": 2, "wireId": 23}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 75]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 23},
        {"index": 1, "wireId": 24},
        {"index": 2, "wireId": 25}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 63]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 26},
        {"index": 1, "wireId": 27},
        {"index": 2, "wireId": 28}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 64]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 28},
        {"index": 1, "wireId": 29},
        {"index": 2, "wireId": 30}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 52]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 31},
        {"index": 1, "wireId": 32},
        {"index": 2, "wireId": 33}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 53]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 33},
        {"index": 1, "wireId": 34},
        {"index": 2, "wireId": 35}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 41]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 36},
        {"index": 1, "wireId": 37},
        {"index": 2, "wireId": 38}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 42]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 38},
        {"index": 1, "wireId": 39},
        {"index": 2, "wireId": 40}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [91, 30]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 41},
        {"index": 1, "wireId": 42},
        {"index": 2, "wireId": 43}
      ]
    },
    {
      "name": "and",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 31]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 43},
        {"index": 1, "wireId": 44},
        {"index": 2, "wireId": 45}
      ]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {
            "name": "translate",
            "args": [105, 107]
          }
        ]
      },
      "connections": [{"index": 0, "wireId": 9}]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [105, 96]}
        ]
      },
      "connections": [{"index": 0, "wireId": 11}]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [105, 85]}
        ]
      },
      "connections": [{"index": 0, "wireId": 16}]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [105, 74]}
        ]
      },
      "connections": [{"index": 0, "wireId": 21}]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [105, 63]}
        ]
      },
      "connections": [{"index": 0, "wireId": 26}]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [105, 52]}
        ]
      },
      "connections": [{"index": 0, "wireId": 31}]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [105, 41]}
        ]
      },
      "connections": [{"index": 0, "wireId": 36}]
    },
    {
      "name": "output",
      "category": "other",
      "transform": {
        "items": [
          {"name": "translate", "args": [105, 30]}
        ]
      },
      "connections": [{"index": 0, "wireId": 41}]
    },
    {
      "name": "not",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [58, 56]},
          {"name": "rotate", "args": ["0", 1, 3]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 30},
        {"index": 0, "wireId": 35},
        {"index": 0, "wireId": 40},
        {"index": 0, "wireId": 45},
        {"index": 1, "wireId": 5}
      ]
    },
    {
      "name": "not",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [55, 43]},
          {"name": "rotate", "args": ["0", 1, 3]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 19},
        {"index": 0, "wireId": 24},
        {"index": 0, "wireId": 39},
        {"index": 0, "wireId": 44},
        {"index": 1, "wireId": 3}
      ]
    },
    {
      "name": "not",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [55, 32]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 46},
        {"index": 0, "wireId": 47},
        {"index": 0, "wireId": 48},
        {"index": 0, "wireId": 49},
        {"index": 1, "wireId": 1}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [55, 87]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 50},
        {"index": 0, "wireId": 51},
        {"index": 0, "wireId": 52},
        {"index": 0, "wireId": 53},
        {"index": 1, "wireId": 0}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [55, 98]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 7},
        {"index": 0, "wireId": 14},
        {"index": 0, "wireId": 29},
        {"index": 0, "wireId": 34},
        {"index": 1, "wireId": 2}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [55, 109]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 8},
        {"index": 0, "wireId": 15},
        {"index": 0, "wireId": 20},
        {"index": 0, "wireId": 25},
        {"index": 1, "wireId": 4}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 27]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 42},
        {"index": 1, "wireId": 46}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 38]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 37},
        {"index": 1, "wireId": 50}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 49]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 32},
        {"index": 1, "wireId": 47}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 60]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 27},
        {"index": 1, "wireId": 51}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 71]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 22},
        {"index": 1, "wireId": 48}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 82]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 17},
        {"index": 1, "wireId": 52}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 93]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 12},
        {"index": 1, "wireId": 49}
      ]
    },
    {
      "name": "repeater",
      "category": "gate",
      "transform": {
        "items": [
          {"name": "translate", "args": [77, 104]}
        ]
      },
      "connections": [
        {"index": 0, "wireId": 10},
        {"index": 1, "wireId": 53}
      ]
    }
  ]
}
```

**Comment by janjaromirhorak** [comment]
Console:
```
aStar: Number of open nodes (100057) exceeded the limit for open nodes (100000).
routeWorker.min.js:1:9077
TypeError: _ is undefined
```

**Comment by janjaromirhorak** [comment]
Preferred behavior: Do not fail when routing of one wire fails. Try to route other wires and than return all wires that have been successfully routed (and return temporary routes for the non routable wires).

**Comment by janjaromirhorak** [comment]
Fixed in 12addf073e1a2176abd6112033568aa760fdb780.
